src/Raven.Server/ServerWide/RawDatabaseRecord.cs

```diff
@@ -16,160 +17,275 @@ namespace Raven.Server.ServerWide
 {
     public class RawDatabaseRecord : IDisposable
     {
-        private readonly BlittableJsonReaderObject _record;
+        private BlittableJsonReaderObject _record;
+
+        private DatabaseRecord _materializedRecord;
```

Comment:
Not sure if possible, but can we do something to prevent mutation of this? and rest of the properties.

Reply:
Not really. We are using fields there. I cannot 'freeze' them.